src/lxc/commands.c

```diff
@@ -506,7 +508,18 @@ static ssize_t lxc_cmd(const char *name, struct lxc_cmd_rr *cmd, bool *stopped,
 
 	*stopped = 0;
 
-	client_fd = lxc_cmd_send(name, cmd, lxcpath, hashed_sock_name);
+	/*
+	 * We don't want to break anything for this special case,
```

Comment:
What is the special case here?

Reply:
we are setting `stay_connected` to `true` right now only for 2 command types `LXC_CMD_GET_TTY_FD` and `LXC_CMD_ADD_STATE_CLIENT`. In this case command socket file descriptor will not be closed after exiting from `lxc_cmd_timeout`. I'm afraid about possible side affects because we have to control how this socket will be used later and that our timeout setup will not break anything. That's why I think it's better not to set timeout for this special cases. Right now we don't have a plan to support timeout for this two command types anyway. But we can easily extend it in the future if needed.

Reply:
Right I see, makes sense. I think a comment there saying something about why those commands are special (what you've explained here) would be valuable for future readers.